masq_lib/src/test_utils/mock_blockchain_client_server.rs

```diff
@@ -0,0 +1,416 @@
+// Copyright (c) 2022, MASQ (https://masq.ai) and/or its affiliates. All rights reserved.
+
+use crate::test_utils::utils::UrlHolder;
+use crate::utils::localhost;
+use crossbeam_channel::{unbounded, Receiver, Sender};
+use itertools::Either;
+use itertools::Either::{Left, Right};
+use lazy_static::lazy_static;
+use regex::Regex;
+use serde::Serialize;
+use std::io::{ErrorKind, Read, Write};
+use std::net::{Ipv4Addr, SocketAddr, SocketAddrV4, TcpListener, TcpStream};
+use std::sync::{Arc, Mutex};
+use std::thread;
+use std::thread::JoinHandle;
+use std::time::Duration;
+
+lazy_static! {
+    pub static ref CONTENT_LENGTH_DETECTOR: Regex =
+        Regex::new(r"[Cc]ontent-[Ll]ength: *(\d+)\r\n").expect("Bad regular expression");
+    pub static ref HTTP_VERSION_DETECTOR: Regex =
+        Regex::new(r"HTTP/(\d\.\d)").expect("Bad regular expression");
+}
+
+pub struct MBCSBuilder {
+    port: u16,
+    run_on_docker: bool,
+    response_batch_opt: Option<Vec<String>>,
+    responses: Vec<String>,
+    notifier: Sender<()>,
+}
+
+impl MBCSBuilder {
+    pub fn new(port: u16) -> Self {
+        Self {
+            port,
+            run_on_docker: false,
+            response_batch_opt: None,
+            responses: vec![],
+            notifier: unbounded().0,
+        }
+    }
+
+    pub fn run_on_docker(mut self) -> Self {
```

Comment:
I'm not sure if I'm in peace with this name. I myself would've chosen `run_in_docker()` as it fits better to my visualization of docker as containers.